swupd/bundleinfo.go

```diff
@@ -66,6 +67,9 @@ func (m *Manifest) GetBundleInfo(stateDir, path string) error {
 			return err
 		}
 		m.BundleInfo.DirectIncludes = includes
+		// TODO: this part of the code seems to be a legacy thing. Keep
+		// an eye on this to make sure m.BundleInfo.OptionalIncludes don't
+		// need to be added here too
```

Comment:
MCA is broken by optional includes. To fix it, I think the following must be done:

1. Set `m.BundleInfo.OptionalIncludes`
2. Update MCA pkg/file subtraction to iterate over the optional includes

Reply:
**`m.BundleInfo.OptionalIncludes`** is already being set up in line 81:
`err = json.Unmarshal(biBytes, &m.BundleInfo)`

when doing the unmarshal of the JSON content the OptionalIncludes will be automatically read from the JSON object and added to the BundleInfo struct.

Reply:
Actually, I was wrong above. We don't want the subtraction.

Reply:
MCA should be happy if the optionally included packages are removed from the AllPackges list.